blog/2020-09-21-setup-bitcoin-cloud-node.md

```diff
@@ -0,0 +1,289 @@
+```
+
+Run `autogen` script:
+
+```shell script
+./autogen.sh`
+```
+
+Run `./configure` script with options:
+
+```shell script
+export BDB_PREFIX='/home/bitcoin/bitcoin/db4'
+./configure --disable-tests --disable-bench --with-gui=no --disable-gui-tests BDB_LIBS="-L${BDB_PREFIX}/lib -ldb_cxx-4.8" BDB_CFLAGS="-I${BDB_PREFIX}/include"
+```
+
+Build bitcoind:
+
+```shell script
+make
+```
+
+Install bitcoind:
+
+```shell script
+sudo make install
+```
+
+### Mounting the harddisk as data dir for bitcoind
+
+For details you can check the [documentation](https://cloud.google.com/compute/docs/disks/add-persistent-disk).
+
+We mount the disk to the [default bitcoind data dir](https://github.com/bitcoin/bitcoin/blob/master/doc/init.md) `/var/lib/bitcoind`.
+
+List disks:
+
+```
+sudo lsblk
+```
+
+Your disk should be named `/dev/sdb` in the list. If not change the following commands accordingly.
+
+Init disk:
+
+```
+sudo mkfs.ext4 -m 0 -F -E lazy_itable_init=0,lazy_journal_init=0,discard /dev/sdb
+```
+
+Mount disk into bitcoind datadir (add to `fstab` so it is automatically done upon system startup):
+
+```
+echo UUID=`sudo blkid -s UUID -o value /dev/sdb` /var/lib/bitcoind ext4 discard,defaults,nofail 0 2 | sudo tee -a /etc/fstab
+```
+
+### Starting bitcoind
+
+As per the [init instructions](https://github.com/bitcoin/bitcoin/blob/master/doc/init.md) for adding bitcoind as service started at instance startup:
+
+Navigate to `contrib/init` dir within the bitcoin repo, you'll find the `bitcoind.service` file there.
+
+Coy said file to register the default configuration of the bitcoin service with systemd:
+
+```
+sudo cp bitcoind.service /lib/systemd/system
+```
+
+Enable the service in systemctl:
+
+```
+sudo systemctl enable bitcoind
+```
+
+Bitcoind is now configured to start automatially at startup of the machine.
+
+You can either restart the machine or just initially start the service manually:
+
+```
+sudo systemctl start bitcoind
+```
+
+The initial block sync will start automatically.
+
+If you are syncing with a slow machine, and your sync seems to be stuck free to add some additional nodes to the configuration, you can find some on [bitnodes.io](https://bitnodes.io). Some nodes have block download limits that can cause this behavior.
+
+Example how to add this to the `bitcoin.conf` (see next section for details):
+
+```
+addnode=34.220.102.44:8333
+addnode=...
+```
+
+### Configure RPC server
+
+Configuration is to be done in the `bitcoin.conf` file, which is to be put into `/etc/bitcoin`. 
+This directory is to be owned by the bitcoin user.
+Make sure to put the configuration file into `/etc/bitcoin/bitcoin.conf` - if file does not exist create it (as user `bitcoin`).
+
+You can check out the official example bitcoin conf [here](https://github.com/bitcoin/bitcoin/blob/master/share/examples/bitcoin.conf).
+
+Make rpc server available and configure user and password for it:
+
+```
+rpcuser=[your username]
+rpcpassword=[your password]
```

Comment:
I'd had a way to securely generate a password using the command line:

`< /dev/urandom tr -dc _A-Z-a-z-0-9 | head -c${1:-32};echo;`

Reply:
:D sweet, but I would leave it up to the user how to do that. 
I usually do this directly in my password manager so I make sure that I use that password